### DISCUSSION TITLE
Cannot install private package, am i doing something wrong ?

### DISCUSSION THREAD

**Discussion by Bilboramix** [pnpm/pnpm#7324] [Q&A]
Hello there, i cannot install my private package with pnpm whereas _**it works well with npm**_ :

Here's the dependency definition in package.json :
```
  "dependencies": {
    "@bilboramix/myPackage": "github:bilboramix/myPackage"
  },
```

.npmrc file :
```
@bilboramix:registry=https://npm.pkg.github.com/
//npm.pkg.github.com/:_authToken=${GITHUB_PACKAGE_TOKEN}
```

Output of the console when i try to do "pnpm install" :

```
Issue while reading "C:\..........\Packaging\myPackage\.npmrc". Failed to replace env in config: ${GITHUB_PACKAGE_TOKEN}
 ERROR  Command failed with exit code 128: git ls-remote git+ssh://git@github.com/bilboramix/myPackage.git HEAD     
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I watched those issues but i can't figure out what i'm doing wrong :
https://github.com/pnpm/pnpm/issues/755
https://github.com/orgs/pnpm/discussions/3930
https://github.com/pnpm/pnpm/issues/2127

I love pnpm and i don't want to fallback to npm :'(

**Comment by Bilboramix** [comment]
the problem was in my package.json :
```
 "dependencies": {
    "@bilboramix/myPackage": "^0.0.0"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Bilboramix/myPackage.git",
  },
```

**Reply by psjadoun** [reply]
how can dependency (source) and repository (consumer) are same package?

**Reply by Bilboramix** [reply]
Tbh i did not know what i was doing. This package.json is very weird☝️ lol
Cannot remember wtf was happening here 🤷‍♂️🤷‍♂️
